src/qtviewer/ParseIfcFile.cpp

```diff
@@ -1,42 +1,169 @@
 #include "ParseIfcFile.h"
+#include "MessageLogger.h"
 
-#include "../ifcgeom_schema_agnostic/IfcGeomIterator.h"
+#include "osg/Array"
+#include "osg/Geometry"
+#include "osg/Material"
+#include "osg/PrimitiveSet"
+#include "osg/StateAttribute"
+#include "osg/StateSet"
+#include "osg/Vec4"
+#include "osg/ref_ptr"
+#include <OpenGL/OpenGL.h>
+#include <cstddef> // size_t
+#include <string>
+
+#include <osg/Vec3>
+#include <vector>
 
 ParseIfcFile::ParseIfcFile() {}
 
 ParseIfcFile::~ParseIfcFile() {}
 
-void ParseIfcFile::outputMsg(const std::string& msg)
-{
-	emit parsingInfo(QString::fromStdString(msg));
-}
-
-void ParseIfcFile::Parse(const std::string& filePath)
-{
+bool ParseIfcFile::Parse(
+    const std::string& filePath,
+    std::vector<osg::ref_ptr<osg::Geometry>>& geometries
+) {
     IfcParse::IfcFile file(filePath);
 
     IfcGeom::IteratorSettings settings;
     settings.set(IfcGeom::IteratorSettings::USE_WORLD_COORDS, true);
     settings.set(IfcGeom::IteratorSettings::WELD_VERTICES, false);
```

Comment:
```suggestion
    settings.set(IfcGeom::IteratorSettings::WELD_VERTICES, false);
```

Sorry, I missed this bit. Setting USE_WORLD_COORDS to true indeed causes the matrix to be the identity matrix.

```
1 0 0 0
0 1 0 0
0 0 1 0
```